src/Disruptor/RingBuffer.cs

```diff
@@ -46,19 +46,19 @@ protected RingBuffer(ISequencer sequencer, Type eventType, int bufferPad)
             {
                 throw new ArgumentException("bufferSize must not be less than 1");
             }
-            if (!_bufferSize.IsPowerOf2())
+            if (!((int)_bufferSize).IsPowerOf2())
             {
                 throw new ArgumentException("bufferSize must be a power of 2");
             }
 
             _entries = Array.CreateInstance(eventType, _bufferSize + 2 * bufferPad);
-            _indexMask = _bufferSize - 1;
+            _indexMask = (int)(_bufferSize - 1);
         }
 
         /// <summary>
         /// Gets the size of the buffer.
         /// </summary>
-        public int BufferSize => _bufferSize;
+        public int BufferSize => (int)_bufferSize;
```

Comment:
Is this cast necessary?

Reply:
No, it left from trying to make the field a long, but here it doesn't matter.